src/build/parse.rs

```diff
@@ -103,90 +103,58 @@ pub fn generate_asts(
+                            stderr.push_str(&err);
                         }
-                        logs::append(package, &err);
-                        has_failure = true;
-                        stderr.push_str(&err);
-                    }
-                };
-                match iast_path {
-                    Ok(Some((_path, err))) => {
+                        _ => (),
+                    };
+                }
+
+                match (iast_path, module.source_type.to_owned()) {
+                    (Ok(Some((_path, Some(err)))), SourceType::SourceFile(ref mut source_file))
+                        if package.is_pinned_dep =>
+                    {
                         // supress warnings in non-pinned deps
-                        if package.is_pinned_dep {
-                            if let Some(err) = err {
-                                match module.source_type {
-                                    SourceType::SourceFile(ref mut source_file) => {
-                                        source_file.interface.as_mut().map(|interface| {
-                                            interface.parse_state = ParseState::ParseError;
-                                            interface.parse_dirty = true;
-                                        });
-                                    }
-                                    _ => (),
-                                }
-                                logs::append(package, &err);
-                                stderr.push_str(&err);
-                            }
+                        if let Some(interface) = source_file.interface.as_mut() {
+                            interface.parse_state = ParseState::Warning;
+                            interface.parse_dirty = true;
                         }
+                        logs::append(package, &err);
+                        stderr.push_str(&err);
                     }
-                    Ok(None) => (),
-                    Err(err) => {
-                        match module.source_type {
-                            SourceType::SourceFile(ref mut source_file) => {
-                                source_file.interface.as_mut().map(|interface| {
-                                    interface.parse_state = ParseState::ParseError;
-                                    interface.parse_dirty = true;
-                                });
-                            }
-                            _ => (),
+                    (Err(err), SourceType::SourceFile(ref mut source_file)) => {
+                        if let Some(interface) = source_file.interface.as_mut() {
+                            interface.parse_state = ParseState::ParseError;
+                            interface.parse_dirty = true;
                         }
                         logs::append(package, &err);
                         has_failure = true;
                         stderr.push_str(&err);
                     }
+                    _ => (),
```

Comment:
beter geen wildcard hier, er zat eerst een bug in omdat we een if clause hadden in een van de arms